File: 70-climbing-stairs/climbing-stairs.py
class Solution(object):
    def rec(self,n,dp):
        if n==0:
            return 1
        if n<0:
            return 0
        if dp[n]!=-1:
            return dp[n]
        sol=self.rec(n-1,dp)+self.rec(n-2,dp)
        dp[n]=sol
        return sol
    def climbStairs(self, n):
        dp=[-1]*(n+1)
        sol=self.rec(n,dp)
        return sol
        

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int nummy(int n,vector<int> &dp){
        if(n==0){
            return 1;
        }
        if(n<0){
            return 0;
        }
        if(dp[n]!=-1){
            return dp[n];
        }
        int sol=nummy(n-1,dp)+nummy(n-2,dp);
        dp[n]=sol;
        return sol;
    }
    int climbStairs(int n) {
        vector<int> dp(n+1,-1);
        int sol=nummy(n,dp);
        return sol;
    }
};